sympy/functions/special/spherical_harmonics.py

```diff
@@ -179,25 +179,28 @@ def fdiff(self, argindex=4):
             raise ArgumentIndexError(self, argindex)
 
     def _eval_rewrite_as_polynomial(self, n, m, theta, phi, **kwargs):
-        # TODO: Make sure n \in N
-        # TODO: Assert |m| <= n ortherwise we should return 0
-        return self.expand(func=True)
+        if n.is_integer and n in S.Naturals:
```

Comment:
```suggestion
        if n in S.Naturals:
```
I think this should be enough? Maybe add some tests to see what happens for different scenarios?